rotkehlchen/chain/bitcoin/manager.py

```diff
@@ -207,3 +281,245 @@ def have_transactions(self, accounts: list[BTCAddress]) -> dict[BTCAddress, tupl
+
+    def query_transactions(
+            self,
+            from_timestamp: Timestamp,
+            to_timestamp: Timestamp,
+            addresses: list[BTCAddress],
+    ) -> None:
+        """Query transactions in the time range for the specified accounts,
+        decode them into HistoryEvents, and save the results to the db.
+        """
+        # TODO: websocket messages for progress updates while querying
+        self.refresh_tracked_accounts()
+        if len(tx_list := self._query(
+            action='transactions',
+            accounts=addresses,
+            options={'from_timestamp': from_timestamp, 'to_timestamp': to_timestamp},
+        )) == 0:
+            return  # no new transactions found
+
+        events = []
+        for tx in tx_list:
+            events.extend(self.decode_transaction(tx))
+
+        with self.database.conn.write_ctx() as write_cursor:
+            DBHistoryEvents(self.database).add_history_events(
+                write_cursor=write_cursor,
+                history=events,
+            )
+
+    @staticmethod
+    def create_event(
+            tx: BitcoinTx,
+            event_type: HistoryEventType,
+            event_subtype: HistoryEventSubType,
+            amount: FVal,
+            notes: str | None = None,
+            location_label: str | None = None,
+    ) -> HistoryEvent:
+        return HistoryEvent(
+            event_identifier=f'{BTC_EVENT_IDENTIFIER_PREFIX}{tx.tx_id}',
+            sequence_index=0,  # events are reshuffled later
+            timestamp=tx.timestamp,
+            location=Location.BITCOIN,
+            event_type=event_type,
+            event_subtype=event_subtype,
+            asset=A_BTC,
+            amount=amount,
+            notes=notes,
+            location_label=location_label,
+        )
+
+    def _maybe_create_fee_events(
+            self,
+            tx: BitcoinTx,
+            totals_per_address: dict[BTCAddress, FVal],
+    ) -> tuple[list[HistoryEvent], dict[BTCAddress, FVal]]:
+        """Create fee events for inputs from tracked addresses.
+        Fee amount per address is proportional to the amount spent by that address.
+        The fee share for each address is subtracted from the total input for that address.
+        Returns the list of fee events and the adjusted input totals in a tuple.
+        """
+        events, total_input = [], sum((per_address_in_totals := totals_per_address).values())
+        for address, amount in per_address_in_totals.items():
+            per_address_in_totals[address] -= (fee_share := (amount / total_input) * tx.fee)
+            if address not in self.tracked_accounts:
+                continue  # only add the fee event for tracked accounts
+
+            events.append(self.create_event(
+                tx=tx,
+                event_type=HistoryEventType.SPEND,
+                event_subtype=HistoryEventSubType.FEE,
+                amount=fee_share,
+                notes=f'Burn {fee_share} BTC for gas',
```

Comment:
BTC has no concept of Gas.

```suggestion
                notes=f'Spend {fee_share} BTC for fees',
```